### PR TITLE
grpc: Use gogoproto for being closer to the the OCI spec

### PR DESCRIPTION
**Pull request by sameo**
With gogoproto, we get a few interesting protobuf extensions for
golang, like the ability to make structure fields non optional and unit
tests generation for the generated code.

This will make the conversion routines between OCI and gRPC faster.

Signed-off-by: Samuel Ortiz <sameo@linux.intel.com>

### CODE REVIEW

**Comment by sboeuf** [comment]
LGTM

File: hack/update-generated-agent-proto.sh [hunk 1]
```diff
@@ -3,4 +3,5 @@
 #
 # SPDX-License-Identifier: Apache-2.0
 #
-protoc --proto_path=protocols/grpc --go_out=plugins=grpc:protocols/grpc protocols/grpc/hyperstart.proto protocols/grpc/oci.proto
+go get -u github.com/gogo/protobuf/gogoproto/
```
Comment (bergwolf) [inline hack/update-generated-agent-proto.sh]: Please do not try to update the binary, which can be time consuming for us.

**Comment by bergwolf** [comment]
Except for the extra step of updating gogoproto, the PR looks good to me.

File: hack/update-generated-agent-proto.sh (hunk 1 above)
Comment (sameo) [reply hack/update-generated-agent-proto.sh]: @bergwolf Done.

**Comment by bergwolf** [comment]
@sameo this now conflicts with master

**Comment by sameo** [comment]
@bergwolf Rebased now.

**Pull request merged**